### ISSUE TITLE
Watchdog Reset Reporting Page

### ISSUE THREAD

**Issue by Hwurzburg** [ArduPilot/ardupilot#15915]
This page is for anyone to report a WatchDog reset (WDG in GCS or DataFlash messages). This allows the Dev team to be alerted, and to investigate as soon as possible.

Please enter the firmware used, vehicle type, and the WDG message. If you have a log, please retain it for 30days in case a Dev would like to examine it.

Note: follow up timing on new issue will depend on many factors, so some may not be addressed immediately.

**Comment by zhaodelong** [comment]
I reported a watchdog reset incident at https://discuss.ardupilot.org/t/arduplane-4-0-9-froze-in-the-middle-of-flight/69140/2

firmware used: arduplane 4.0.9 stable,
vehicle type: plane
the WDG message: decoded message in the forum link

Could you investigate this? logs and videos in the forum

**Comment by tridge** [comment]
This one from Simon Exner 
https://discuss.ardupilot.org/t/watchdog-reset-in-flight-and-ardupilot-does-not-complete-startup-and-hangs/64447/5

**Comment by giacomo892** [comment]
Copter 1fe1e532607de07a38ff9eed44dbb3ae9d333a64 
Matek F405-STD
![image](https://user-images.githubusercontent.com/20860004/114668157-b0667e80-9d00-11eb-877e-fd6aa9aac207.png)

WDG: T-2 SL0 FL0 FT0 FA0 FTP0 FLR0 FICSR0 MM0 MC0

I think it watchdogged moments later I've switched to auto

Can share logs with EKF_REPLAY enabled.

```
ACRO_BAL_PITCH   1.000000
ACRO_BAL_ROLL    1.000000
ACRO_OPTIONS     0
ACRO_RP_EXPO     0.300000
ACRO_RP_P        4.700000
ACRO_THR_MID     0.000000
ACRO_TRAINER     2
ACRO_YAW_P       3.500000
ACRO_Y_EXPO      0.000000
AHRS_COMP_BETA   0.100000
AHRS_CUSTOM_PIT  0.000000
AHRS_CUSTOM_ROLL 0.000000
AHRS_CUSTOM_YAW  0.000000
AHRS_EKF_TYPE    2
AHRS_GPS_GAIN    1.000000
AHRS_GPS_MINSATS 6
AHRS_GPS_USE     1
AHRS_ORIENTATION 0
AHRS_RP_P        0.200000
AHRS_TRIM_X      0.000000
AHRS_TRIM_Y      0.010000
AHRS_TRIM_Z      0.000000
AHRS_WIND_MAX    0
AHRS_YAW_P       0.200000
ANGLE_MAX        3000
ARMING_ACCTHRESH 0.750000
ARMING_CHECK     1
ARMING_MIS_ITEMS 0
ARMING_RUDDER    2
ATC_ACCEL_P_MAX  110000.000000
ATC_ACCEL_R_MAX  110000.000000
ATC_ACCEL_Y_MAX  19000.000000
ATC_ANGLE_BOOST  1
ATC_ANG_LIM_TC   1.000000
ATC_ANG_PIT_P    18.000000
ATC_ANG_RLL_P    16.200001
ATC_ANG_YAW_P    4.300000
ATC_INPUT_TC     0.150000
ATC_RATE_FF_ENAB 1
ATC_RATE_P_MAX   0.000000
ATC_RATE_R_MAX   0.000000
ATC_RATE_Y_MAX   0.000000
ATC_RAT_PIT_D    0.007500
ATC_RAT_PIT_FF   0.000000
ATC_RAT_PIT_FLTD 20.000000
ATC_RAT_PIT_FLTE 0.000000
ATC_RAT_PIT_FLTT 25.000000
ATC_RAT_PIT_I    0.102661
ATC_RAT_PIT_IMAX 0.500000
ATC_RAT_PIT_P    0.110000
ATC_RAT_PIT_SMAX 0.000000
ATC_RAT_RLL_D    0.006500
ATC_RAT_RLL_FF   0.000000
ATC_RAT_RLL_FLTD 20.000000
ATC_RAT_RLL_FLTE 0.000000
ATC_RAT_RLL_FLTT 25.000000
ATC_RAT_RLL_I    0.079862
ATC_RAT_RLL_IMAX 0.500000
ATC_RAT_RLL_P    0.080000
ATC_RAT_RLL_SMAX 0.000000
ATC_RAT_YAW_D    0.000000
ATC_RAT_YAW_FF   0.000000
ATC_RAT_YAW_FLTD 0.000000
ATC_RAT_YAW_FLTE 2.000000
ATC_RAT_YAW_FLTT 25.000000
ATC_RAT_YAW_I    0.042000
ATC_RAT_YAW_IMAX 0.500000
ATC_RAT_YAW_P    0.490000
ATC_RAT_YAW_SMAX 0.000000
ATC_SLEW_YAW     6000.000000
ATC_THR_MIX_MAN  0.300000
ATC_THR_MIX_MAX  0.900000
ATC_THR_MIX_MIN  0.250000
AUTOTUNE_AGGR    0.100000
AUTOTUNE_AXES    7
AUTOTUNE_MIN_D   0.001000
AUTO_OPTIONS     0
AVOID_ACCEL_MAX  3.000000
AVOID_ALT_MIN    0.000000
AVOID_ANGLE_MAX  1000
AVOID_BACKUP_SPD 0.750000
AVOID_BEHAVE     0
AVOID_DIST_MAX   5.000000
AVOID_ENABLE     3
AVOID_MARGIN     2.000000
BARO1_DEVID      816897
BARO1_GND_PRESS  92540.085938
BARO2_DEVID      0
BARO2_GND_PRESS  0.000000
BARO3_DEVID      0
BARO3_GND_PRESS  0.000000
BARO_ALT_OFFSET  0.000000
BARO_EXT_BUS     -1
BARO_FLTR_RNG    0
BARO_GND_TEMP    0.000000
BARO_PRIMARY     0
BARO_PROBE_EXT   0
BATT2_MONITOR    0
BATT3_MONITOR    0
BATT4_MONITOR    0
BATT5_MONITOR    0
BATT6_MONITOR    0
BATT7_MONITOR    0
BATT8_MONITOR    0
BATT9_MONITOR    0
BATT_AMP_OFFSET  0.003200
BATT_AMP_PERVLT  54.599998
BATT_ARM_MAH     0
BATT_ARM_VOLT    0.000000
BATT_BUS         0
BATT_CAPACITY    3300
BATT_CRT_MAH     0.000000
BATT_CRT_VOLT    0.000000
BATT_CURR_PIN    14
BATT_FS_CRT_ACT  0
BATT_FS_LOW_ACT  0
BATT_FS_VOLTSRC  0
BATT_LOW_MAH     0.000000
BATT_LOW_TIMER   10
BATT_LOW_VOLT    10.500000
BATT_MONITOR     4
BATT_OPTIONS     0
BATT_SERIAL_NUM  -1
BATT_VOLT_MULT   10.973381
BATT_VOLT_PIN    15
BRD_ALT_CONFIG   0
BRD_BOOT_DELAY   0
BRD_OPTIONS      1
BRD_PWM_COUNT    5
BRD_RTC_TYPES    1
BRD_RTC_TZ_MIN   0
BRD_SAFETYOPTION 3
BRD_SD_SLOWDOWN  0
BRD_SERIAL_NUM   0
BTN_ENABLE       0
CAM_AUTO_ONLY    0
CAM_DURATION     10
CAM_FEEDBACK_PIN -1
CAM_FEEDBACK_POL 1
CAM_MAX_ROLL     0
CAM_MIN_INTERVAL 0
CAM_RELAY_ON     1
CAM_SERVO_OFF    1100
CAM_SERVO_ON     1300
CAM_TRIGG_DIST   0.000000
CAM_TRIGG_TYPE   0
CAM_TYPE         0
CIRCLE_OPTIONS   13
CIRCLE_RADIUS    1000.000000
CIRCLE_RATE      90.000000
COMPASS_AUTODEC  1
COMPASS_AUTO_ROT 2
COMPASS_CAL_FIT  100.000000
COMPASS_CUS_PIT  0.000000
COMPASS_CUS_ROLL 0.000000
COMPASS_CUS_YAW  0.000000
COMPASS_DEC      0.000000
COMPASS_DEV_ID   466433
COMPASS_DEV_ID2  0
COMPASS_DEV_ID3  0
COMPASS_DEV_ID4  0
COMPASS_DEV_ID5  0
COMPASS_DEV_ID6  0
COMPASS_DEV_ID7  0
COMPASS_DEV_ID8  0
COMPASS_DIA2_X   0.000000
COMPASS_DIA2_Y   0.000000
COMPASS_DIA2_Z   0.000000
COMPASS_DIA3_X   0.000000
COMPASS_DIA3_Y   0.000000
COMPASS_DIA3_Z   0.000000
COMPASS_DIA_X    1.002000
COMPASS_DIA_Y    0.985000
COMPASS_DIA_Z    1.013000
COMPASS_ENABLE   1
COMPASS_EXTERN2  0
COMPASS_EXTERN3  0
COMPASS_EXTERNAL 1
COMPASS_FLTR_RNG 0
COMPASS_LEARN    0
COMPASS_MOT2_X   0.000000
COMPASS_MOT2_Y   0.000000
COMPASS_MOT2_Z   0.000000
COMPASS_MOT3_X   0.000000
COMPASS_MOT3_Y   0.000000
COMPASS_MOT3_Z   0.000000
COMPASS_MOTCT    0
COMPASS_MOT_X    0.000000
COMPASS_MOT_Y    0.000000
COMPASS_MOT_Z    0.000000
COMPASS_ODI2_X   0.000000
COMPASS_ODI2_Y   0.000000
COMPASS_ODI2_Z   0.000000
COMPASS_ODI3_X   0.000000
COMPASS_ODI3_Y   0.000000
COMPASS_ODI3_Z   0.000000
COMPASS_ODI_X    -0.008011
COMPASS_ODI_Y    0.001471
COMPASS_ODI_Z    0.012284
COMPASS_OFFS_MAX 1800
COMPASS_OFS2_X   0.000000
COMPASS_OFS2_Y   0.000000
COMPASS_OFS2_Z   0.000000
COMPASS_OFS3_X   0.000000
COMPASS_OFS3_Y   0.000000
COMPASS_OFS3_Z   0.000000
COMPASS_OFS_X    -83.000000
COMPASS_OFS_Y    2.000000
COMPASS_OFS_Z    40.000000
COMPASS_OPTIONS  0
COMPASS_ORIENT   0
COMPASS_ORIENT2  0
COMPASS_ORIENT3  0
COMPASS_PMOT_EN  0
COMPASS_PRIO1_ID 466433
COMPASS_PRIO2_ID 0
COMPASS_PRIO3_ID 0
COMPASS_SCALE    1.080000
COMPASS_SCALE2   0.000000
COMPASS_SCALE3   0.000000
COMPASS_TYPEMASK 0
COMPASS_USE      1
COMPASS_USE2     1
COMPASS_USE3     1
DEV_OPTIONS      0
DISARM_DELAY     20
EK2_ABIAS_P_NSE  0.005000
EK2_ACC_P_NSE    0.600000
EK2_ALT_M_NSE    2.000000
EK2_ALT_SOURCE   0
EK2_BCN_DELAY    50
EK2_BCN_I_GTE    500
EK2_BCN_M_NSE    1.000000
EK2_CHECK_SCALE  100
EK2_EAS_I_GATE   400
EK2_EAS_M_NSE    1.400000
EK2_ENABLE       1
EK2_FLOW_DELAY   10
EK2_FLOW_I_GATE  300
EK2_FLOW_M_NSE   0.250000
EK2_FLOW_USE     1
EK2_GBIAS_P_NSE  0.000100
EK2_GLITCH_RAD   25
EK2_GPS_CHECK    31
EK2_GPS_TYPE     0
EK2_GSCL_P_NSE   0.000500
EK2_GSF_DELAY    1000
EK2_GSF_RST_MAX  2
EK2_GSF_RUN_MASK 3
EK2_GSF_USE_MASK 3
EK2_GYRO_P_NSE   0.030000
EK2_HGT_DELAY    30
EK2_HGT_I_GATE   500
EK2_HRT_FILT     2.000000
EK2_IMU_MASK     1
EK2_MAGB_P_NSE   0.000100
EK2_MAGE_P_NSE   0.001000
EK2_MAG_CAL      3
EK2_MAG_EF_LIM   50
EK2_MAG_I_GATE   300
EK2_MAG_MASK     0
EK2_MAG_M_NSE    0.050000
EK2_MAX_FLOW     2.500000
EK2_NOAID_M_NSE  10.000000
EK2_OGN_HGT_MASK 0
EK2_POSNE_M_NSE  1.000000
EK2_POS_I_GATE   500
EK2_RNG_I_GATE   500
EK2_RNG_M_NSE    0.500000
EK2_RNG_USE_HGT  -1
EK2_RNG_USE_SPD  2.000000
EK2_TAU_OUTPUT   25
EK2_TERR_GRAD    0.100000
EK2_VELD_M_NSE   0.500000
EK2_VELNE_M_NSE  0.300000
EK2_VEL_I_GATE   500
EK2_WIND_PSCALE  0.500000
EK2_WIND_P_NSE   0.100000
EK2_YAW_I_GATE   300
EK2_YAW_M_NSE    0.500000
ESC_CALIBRATION  0
FENCE_ACTION     1
FENCE_ALT_MAX    100.000000
FENCE_ALT_MIN    -10.000000
FENCE_ENABLE     0
FENCE_MARGIN     2.000000
FENCE_RADIUS     300.000000
FENCE_TOTAL      0
FENCE_TYPE       7
FLIGHT_OPTIONS   0
FLOW_ADDR        0
FLOW_FXSCALER    0
FLOW_FYSCALER    0
FLOW_ORIENT_YAW  0
FLOW_POS_X       0.000000
FLOW_POS_Y       0.000000
FLOW_POS_Z       0.000000
FLOW_TYPE        0
FLTMODE1         0
FLTMODE2         5
FLTMODE3         6
FLTMODE4         3
FLTMODE5         11
FLTMODE6         2
FLTMODE_CH       7
FORMAT_VERSION   120
FRAME_CLASS      1
FRAME_TYPE       1
FRSKY_DNLINK1_ID 20
FRSKY_DNLINK2_ID 7
FRSKY_UPLINK_ID  13
FS_CRASH_CHECK   1
FS_EKF_ACTION    1
FS_EKF_THRESH    0.800000
FS_GCS_ENABLE    0
FS_GCS_TIMEOUT   5.000000
FS_OPTIONS       0
FS_THR_ENABLE    1
FS_THR_VALUE     975
FS_VIBE_ENABLE   1
GCS_PID_MASK     0
GND_EFFECT_COMP  1
GPS_AUTO_CONFIG  1
GPS_AUTO_SWITCH  1
GPS_BLEND_MASK   5
GPS_BLEND_TC     10.000000
GPS_COM_PORT     1
GPS_COM_PORT2    1
GPS_DELAY_MS     0
GPS_DELAY_MS2    0
GPS_GNSS_MODE    71
GPS_GNSS_MODE2   0
GPS_HDOP_GOOD    140
GPS_INJECT_TO    127
GPS_MIN_DGPS     100
GPS_MIN_ELEV     20
GPS_NAVFILTER    8
GPS_POS1_X       0.080000
GPS_POS1_Y       0.080000
GPS_POS1_Z       -0.013000
GPS_POS2_X       0.000000
GPS_POS2_Y       0.000000
GPS_POS2_Z       0.000000
GPS_PRIMARY      0
GPS_RATE_MS      100
GPS_RATE_MS2     200
GPS_RAW_DATA     0
GPS_SAVE_CFG     0
GPS_SBAS_MODE    1
GPS_SBP_LOGMASK  -256
GPS_TYPE         1
GPS_TYPE2        0
GUID_OPTIONS     0
INITIAL_MODE     0
INS_ACCEL_FILTER 15
INS_ACCOFFS_X    0.059628
INS_ACCOFFS_Y    -0.159460
INS_ACCOFFS_Z    -0.085176
INS_ACCSCAL_X    0.997067
INS_ACCSCAL_Y    1.016713
INS_ACCSCAL_Z    0.997627
INS_ACC_BODYFIX  2
INS_ACC_ID       3080458
INS_ENABLE_MASK  127
INS_FAST_SAMPLE  1
INS_GYROFFS_X    -0.012360
INS_GYROFFS_Y    -0.001803
INS_GYROFFS_Z    -0.011766
INS_GYRO_FILTER  50
INS_GYRO_RATE    1
INS_GYR_CAL      1
INS_GYR_ID       3080458
INS_HNTCH_ENABLE 0
INS_LOG_BAT_CNT  1024
INS_LOG_BAT_LGCT 32
INS_LOG_BAT_LGIN 20
INS_LOG_BAT_MASK 0
INS_LOG_BAT_OPT  0
INS_NOTCH_ENABLE 0
INS_POS1_X       0.000000
INS_POS1_Y       0.000000
INS_POS1_Z       0.000000
INS_STILL_THRESH 2.500000
INS_TRIM_OPTION  1
INS_USE          1
LAND_ALT_LOW     1000
LAND_REPOSITION  1
LAND_SPEED       50
LAND_SPEED_HIGH  0
LGR_DEPLOY_ALT   0
LGR_DEPLOY_PIN   -1
LGR_DEPLOY_POL   0
LGR_OPTIONS      3
LGR_RETRACT_ALT  0
LGR_STARTUP      0
LGR_WOW_PIN      -1
LGR_WOW_POL      0
LOG_BACKEND_TYPE 1
LOG_BITMASK      65534
LOG_DISARMED     1
LOG_FILE_BUFSIZE 16
LOG_FILE_DSRMROT 0
LOG_FILE_MB_FREE 500
LOG_FILE_TIMEOUT 5
LOG_MAV_BUFSIZE  8
LOG_REPLAY       1
LOIT_ACC_MAX     500.000000
LOIT_ANG_MAX     0.000000
LOIT_BRK_ACCEL   350.000000
LOIT_BRK_DELAY   0.100000
LOIT_BRK_JERK    500.000000
LOIT_SPEED       1400.000000
MIS_OPTIONS      0
MIS_RESTART      0
MIS_TOTAL        6
MNT_ANGMAX_PAN   4500
MNT_ANGMAX_ROL   4500
MNT_ANGMAX_TIL   4500
MNT_ANGMIN_PAN   -4500
MNT_ANGMIN_ROL   -4500
MNT_ANGMIN_TIL   -4500
MNT_DEFLT_MODE   3
MNT_JSTICK_SPD   0
MNT_LEAD_PTCH    0.200000
MNT_LEAD_RLL     0.000000
MNT_NEUTRAL_X    0.000000
MNT_NEUTRAL_Y    0.000000
MNT_NEUTRAL_Z    0.000000
MNT_RC_IN_PAN    0
MNT_RC_IN_ROLL   0
MNT_RC_IN_TILT   0
MNT_RETRACT_X    0.000000
MNT_RETRACT_Y    0.000000
MNT_RETRACT_Z    0.000000
MNT_STAB_PAN     0
MNT_STAB_ROLL    0
MNT_STAB_TILT    1
MNT_TYPE         1
MOT_BAT_CURR_MAX 0.000000
MOT_BAT_CURR_TC  5.000000
MOT_BAT_IDX      0
MOT_BAT_VOLT_MAX 17.500000
MOT_BAT_VOLT_MIN 13.800000
MOT_BOOST_SCALE  0.000000
MOT_HOVER_LEARN  2
MOT_PWM_MAX      2000
MOT_PWM_MIN      1000
MOT_PWM_TYPE     4
MOT_SAFE_DISARM  0
MOT_SAFE_TIME    1.000000
MOT_SLEW_DN_TIME 0.000000
MOT_SLEW_UP_TIME 0.000000
MOT_SPIN_ARM     0.129000
MOT_SPIN_MAX     0.950000
MOT_SPIN_MIN     0.160000
MOT_SPOOL_TIME   2.000000
MOT_THST_EXPO    0.650000
MOT_THST_HOVER   0.148868
MOT_YAW_HEADROOM 200
NTF_BUZZ_ENABLE  1
NTF_BUZZ_VOLUME  100
NTF_DISPLAY_TYPE 0
NTF_LED_BRIGHT   3
NTF_LED_LEN      1
NTF_LED_OVERRIDE 0
NTF_LED_TYPES    199
OSD1_ALTITUDE_EN 1
OSD1_ALTITUDE_X  23
OSD1_ALTITUDE_Y  8
OSD1_ASPD1_EN    0
OSD1_ASPD1_X     0
OSD1_ASPD1_Y     0
OSD1_ASPD2_EN    0
OSD1_ASPD2_X     0
OSD1_ASPD2_Y     0
OSD1_ASPEED_EN   0
OSD1_ASPEED_X    2
OSD1_ASPEED_Y    13
OSD1_ATEMP_EN    0
OSD1_ATEMP_X     0
OSD1_ATEMP_Y     0
OSD1_AVGCELLV_EN 0
OSD1_AVGCELLV_X  24
OSD1_AVGCELLV_Y  3
OSD1_BAT2USED_EN 0
OSD1_BAT2USED_X  0
OSD1_BAT2USED_Y  0
OSD1_BAT2_VLT_EN 0
OSD1_BAT2_VLT_X  0
OSD1_BAT2_VLT_Y  0
OSD1_BATUSED_EN  1
OSD1_BATUSED_X   23
OSD1_BATUSED_Y   3
OSD1_BAT_VOLT_EN 1
OSD1_BAT_VOLT_X  24
OSD1_BAT_VOLT_Y  1
OSD1_BLHAMPS_EN  0
OSD1_BLHAMPS_X   24
OSD1_BLHAMPS_Y   14
OSD1_BLHRPM_EN   1
OSD1_BLHRPM_X    23
OSD1_BLHRPM_Y    4
OSD1_BLHTEMP_EN  1
OSD1_BLHTEMP_X   25
OSD1_BLHTEMP_Y   5
OSD1_BTEMP_EN    0
OSD1_BTEMP_X     0
OSD1_BTEMP_Y     0
OSD1_CALLSIGN_EN 0
OSD1_CALLSIGN_X  0
OSD1_CALLSIGN_Y  0
OSD1_CHAN_MAX    2100
OSD1_CHAN_MIN    900
OSD1_CLIMBEFF_EN 0
OSD1_CLIMBEFF_X  0
OSD1_CLIMBEFF_Y  0
OSD1_CLK_EN      0
OSD1_CLK_X       0
OSD1_CLK_Y       0
OSD1_COMPASS_EN  0
OSD1_COMPASS_X   15
OSD1_COMPASS_Y   3
OSD1_CURRENT2_EN 0
OSD1_CURRENT2_X  0
OSD1_CURRENT2_Y  0
OSD1_CURRENT_EN  1
OSD1_CURRENT_X   25
OSD1_CURRENT_Y   2
OSD1_DIST_EN     0
OSD1_DIST_X      18
OSD1_DIST_Y      3
OSD1_EFF_EN      0
OSD1_EFF_X       22
OSD1_EFF_Y       10
OSD1_ENABLE      1
OSD1_FLTIME_EN   1
OSD1_FLTIME_X    21
OSD1_FLTIME_Y    11
OSD1_FLTMODE_EN  1
OSD1_FLTMODE_X   2
OSD1_FLTMODE_Y   8
OSD1_GPSLAT_EN   1
OSD1_GPSLAT_X    9
OSD1_GPSLAT_Y    12
OSD1_GPSLONG_EN  1
OSD1_GPSLONG_X   9
OSD1_GPSLONG_Y   13
OSD1_GSPEED_EN   1
OSD1_GSPEED_X    2
OSD1_GSPEED_Y    12
OSD1_HDOP_EN     1
OSD1_HDOP_X      6
OSD1_HDOP_Y      2
OSD1_HEADING_EN  1
OSD1_HEADING_X   13
OSD1_HEADING_Y   2
OSD1_HOME_EN     1
OSD1_HOME_X      14
OSD1_HOME_Y      1
OSD1_HORIZON_EN  1
OSD1_HORIZON_X   14
OSD1_HORIZON_Y   8
OSD1_MESSAGE_EN  1
OSD1_MESSAGE_X   2
OSD1_MESSAGE_Y   6
OSD1_PITCH_EN    1
OSD1_PITCH_X     15
OSD1_PITCH_Y     10
OSD1_RESTVOLT_EN 0
OSD1_RESTVOLT_X  24
OSD1_RESTVOLT_Y  2
OSD1_ROLL_EN     1
OSD1_ROLL_X      9
OSD1_ROLL_Y      10
OSD1_RSSI_EN     1
OSD1_RSSI_X      1
OSD1_RSSI_Y      1
OSD1_SATS_EN     1
OSD1_SATS_X      1
OSD1_SATS_Y      2
OSD1_STATS_EN    0
OSD1_STATS_X     0
OSD1_STATS_Y     0
OSD1_TEMP_EN     0
OSD1_TEMP_X      0
OSD1_TEMP_Y      0
OSD1_TER_HGT_EN  1
OSD1_TER_HGT_X   23
OSD1_TER_HGT_Y   7
OSD1_THROTTLE_EN 1
OSD1_THROTTLE_X  24
OSD1_THROTTLE_Y  12
OSD1_VSPEED_EN   1
OSD1_VSPEED_X    24
OSD1_VSPEED_Y    9
OSD1_VTX_PWR_EN  0
OSD1_VTX_PWR_X   0
OSD1_VTX_PWR_Y   0
OSD1_WAYPOINT_EN 0
OSD1_WAYPOINT_X  0
OSD1_WAYPOINT_Y  0
OSD1_WIND_EN     0
OSD1_WIND_X      2
OSD1_WIND_Y      12
OSD1_XTRACK_EN   0
OSD1_XTRACK_X    0
OSD1_XTRACK_Y    0
OSD2_ENABLE      0
OSD3_ENABLE      0
OSD4_ENABLE      0
OSD_ARM_SCR      0
OSD_CELL_COUNT   -1
OSD_CHAN         0
OSD_DSARM_SCR    0
OSD_FONT         0
OSD_FS_SCR       0
OSD_H_OFFSET     32
OSD_MSG_TIME     10
OSD_OPTIONS      1
OSD_SW_METHOD    0
OSD_TYPE         1
OSD_UNITS        0
OSD_V_OFFSET     16
OSD_W_AVGCELLV   3.600000
OSD_W_BATVOLT    10.000000
OSD_W_NSAT       9
OSD_W_RESTVOLT   10.000000
OSD_W_RSSI       30
OSD_W_TERR       -1
PHLD_BRAKE_ANGLE 2500
PHLD_BRAKE_RATE  12
PILOT_ACCEL_Z    550
PILOT_SPEED_DN   0
PILOT_SPEED_UP   300
PILOT_THR_BHV    0
PILOT_THR_FILT   0.000000
PILOT_TKOFF_ALT  0.000000
PSC_ACCZ_D       0.000000
PSC_ACCZ_FF      0.000000
PSC_ACCZ_FLTD    0.000000
PSC_ACCZ_FLTE    20.000000
PSC_ACCZ_FLTT    0.000000
PSC_ACCZ_I       0.290000
PSC_ACCZ_IMAX    800.000000
PSC_ACCZ_P       0.150000
PSC_ACCZ_SMAX    0.000000
PSC_ACC_XY_FILT  2.000000
PSC_ANGLE_MAX    0.000000
PSC_POSXY_P      1.000000
PSC_POSZ_P       1.000000
PSC_VELXY_D      0.500000
PSC_VELXY_D_FILT 5.000000
PSC_VELXY_FF     0.000000
PSC_VELXY_FILT   5.000000
PSC_VELXY_I      1.000000
PSC_VELXY_IMAX   1000.000000
PSC_VELXY_P      2.000000
PSC_VELZ_D       0.000000
PSC_VELZ_FF      0.000000
PSC_VELZ_FLTD    5.000000
PSC_VELZ_FLTE    5.000000
PSC_VELZ_I       0.000000
PSC_VELZ_IMAX    1000.000000
PSC_VELZ_P       5.000000
RALLY_INCL_HOME  1
RALLY_LIMIT_KM   0.300000
RALLY_TOTAL      0
RC10_DZ          0
RC10_MAX         2003
RC10_MIN         1024
RC10_OPTION      0
RC10_REVERSED    0
RC10_TRIM        1024
RC11_DZ          0
RC11_MAX         2000
RC11_MIN         1000
RC11_OPTION      0
RC11_REVERSED    0
RC11_TRIM        1994
RC12_DZ          0
RC12_MAX         2000
RC12_MIN         800
RC12_OPTION      0
RC12_REVERSED    0
RC12_TRIM        874
RC13_DZ          0
RC13_MAX         2000
RC13_MIN         800
RC13_OPTION      0
RC13_REVERSED    0
RC13_TRIM        874
RC14_DZ          0
RC14_MAX         2000
RC14_MIN         800
RC14_OPTION      0
RC14_REVERSED    0
RC14_TRIM        874
RC15_DZ          0
RC15_MAX         2000
RC15_MIN         800
RC15_OPTION      0
RC15_REVERSED    0
RC15_TRIM        874
RC16_DZ          0
RC16_MAX         2000
RC16_MIN         800
RC16_OPTION      0
RC16_REVERSED    0
RC16_TRIM        874
RC1_DZ           20
RC1_MAX          2003
RC1_MIN          1024
RC1_OPTION       0
RC1_REVERSED     0
RC1_TRIM         1497
RC2_DZ           20
RC2_MAX          1998
RC2_MIN          1029
RC2_OPTION       0
RC2_REVERSED     1
RC2_TRIM         1497
RC3_DZ           35
RC3_MAX          1994
RC3_MIN          1000
RC3_OPTION       0
RC3_REVERSED     0
RC3_TRIM         1000
RC4_DZ           20
RC4_MAX          2003
RC4_MIN          1024
RC4_OPTION       0
RC4_REVERSED     0
RC4_TRIM         1497
RC5_DZ           0
RC5_MAX          1998
RC5_MIN          1004
RC5_OPTION       72
RC5_REVERSED     0
RC5_TRIM         1500
RC6_DZ           0
RC6_MAX          1998
RC6_MIN          1004
RC6_OPTION       4
RC6_REVERSED     0
RC6_TRIM         1004
RC7_DZ           0
RC7_MAX          1998
RC7_MIN          1004
RC7_OPTION       0
RC7_REVERSED     0
RC7_TRIM         1004
RC8_DZ           0
RC8_MAX          1998
RC8_MIN          1004
RC8_OPTION       0
RC8_REVERSED     0
RC8_TRIM         1004
RC9_DZ           0
RC9_MAX          2003
RC9_MIN          1024
RC9_OPTION       0
RC9_REVERSED     0
RC9_TRIM         1024
RCMAP_PITCH      2
RCMAP_ROLL       1
RCMAP_THROTTLE   3
RCMAP_YAW        4
RC_OPTIONS       0
RC_OVERRIDE_TIME 3.000000
RC_PROTOCOLS     1
RC_SPEED         490
RELAY_DEFAULT    0
RELAY_PIN        -1
RELAY_PIN2       -1
RELAY_PIN3       -1
RELAY_PIN4       -1
RELAY_PIN5       -1
RELAY_PIN6       -1
RNGFND1_TYPE     0
RNGFND2_TYPE     0
RNGFND3_TYPE     0
RNGFND4_TYPE     0
RNGFND5_TYPE     0
RNGFND6_TYPE     0
RNGFND7_TYPE     0
RNGFND8_TYPE     0
RNGFND9_TYPE     0
RNGFNDA_TYPE     0
RNGFND_GAIN      0.800000
RSSI_ANA_PIN     9
RSSI_CHANNEL     15
RSSI_CHAN_HIGH   2000
RSSI_CHAN_LOW    1000
RSSI_PIN_HIGH    5.000000
RSSI_PIN_LOW     0.000000
RSSI_TYPE        2
RTL_ALT          1500
RTL_ALT_FINAL    0
RTL_ALT_TYPE     0
RTL_CLIMB_MIN    0
RTL_CONE_SLOPE   3.000000
RTL_LOIT_TIME    2000
RTL_OPTIONS      0
RTL_SPEED        1000
SCHED_DEBUG      0
SCHED_LOOP_RATE  500
SCHED_OPTIONS    1
SERIAL0_BAUD     115
SERIAL0_PROTOCOL 2
SERIAL1_BAUD     115200
SERIAL1_OPTIONS  0
SERIAL1_PROTOCOL 16
SERIAL2_BAUD     57
SERIAL2_OPTIONS  0
SERIAL2_PROTOCOL -1
SERIAL3_BAUD     115
SERIAL3_OPTIONS  0
SERIAL3_PROTOCOL 5
SERIAL4_BAUD     921
SERIAL4_OPTIONS  0
SERIAL4_PROTOCOL 2
SERIAL5_BAUD     57
SERIAL5_OPTIONS  4
SERIAL5_PROTOCOL 10
SERIAL_PASS1     0
SERIAL_PASS2     -1
SERIAL_PASSTIMO  30
SERVO10_FUNCTION 0
SERVO10_MAX      1900
SERVO10_MIN      1100
SERVO10_REVERSED 0
SERVO10_TRIM     1500
SERVO11_FUNCTION 0
SERVO11_MAX      1900
SERVO11_MIN      1100
SERVO11_REVERSED 0
SERVO11_TRIM     1500
SERVO12_FUNCTION 0
SERVO12_MAX      1900
SERVO12_MIN      1100
SERVO12_REVERSED 0
SERVO12_TRIM     1500
SERVO13_FUNCTION 0
SERVO13_MAX      1900
SERVO13_MIN      1100
SERVO13_REVERSED 0
SERVO13_TRIM     1500
SERVO14_FUNCTION 0
SERVO14_MAX      1900
SERVO14_MIN      1100
SERVO14_REVERSED 0
SERVO14_TRIM     1500
SERVO15_FUNCTION 0
SERVO15_MAX      1900
SERVO15_MIN      1100
SERVO15_REVERSED 0
SERVO15_TRIM     1500
SERVO16_FUNCTION 0
SERVO16_MAX      1900
SERVO16_MIN      1100
SERVO16_REVERSED 0
SERVO16_TRIM     1500
SERVO1_FUNCTION  34
SERVO1_MAX       2000
SERVO1_MIN       1000
SERVO1_REVERSED  0
SERVO1_TRIM      1000
SERVO2_FUNCTION  36
SERVO2_MAX       2000
SERVO2_MIN       1000
SERVO2_REVERSED  0
SERVO2_TRIM      1000
SERVO3_FUNCTION  35
SERVO3_MAX       2000
SERVO3_MIN       1000
SERVO3_REVERSED  0
SERVO3_TRIM      1000
SERVO4_FUNCTION  33
SERVO4_MAX       2000
SERVO4_MIN       1000
SERVO4_REVERSED  0
SERVO4_TRIM      1000
SERVO5_FUNCTION  7
SERVO5_MAX       1900
SERVO5_MIN       1100
SERVO5_REVERSED  0
SERVO5_TRIM      1500
SERVO6_FUNCTION  0
SERVO6_MAX       1900
SERVO6_MIN       1100
SERVO6_REVERSED  0
SERVO6_TRIM      1500
SERVO7_FUNCTION  0
SERVO7_MAX       1900
SERVO7_MIN       1100
SERVO7_REVERSED  0
SERVO7_TRIM      1500
SERVO8_FUNCTION  0
SERVO8_MAX       1900
SERVO8_MIN       1100
SERVO8_REVERSED  0
SERVO8_TRIM      1500
SERVO9_FUNCTION  0
SERVO9_MAX       1900
SERVO9_MIN       1100
SERVO9_REVERSED  0
SERVO9_TRIM      1500
SERVO_BLH_AUTO   1
SERVO_BLH_DEBUG  0
SERVO_BLH_MASK   0
SERVO_BLH_OTYPE  0
SERVO_BLH_POLES  14
SERVO_BLH_PORT   0
SERVO_BLH_REMASK 0
SERVO_BLH_TEST   0
SERVO_BLH_TMOUT  0
SERVO_BLH_TRATE  10
SERVO_DSHOT_RATE 3
SERVO_RATE       50
SERVO_ROB_POSMAX 4095
SERVO_ROB_POSMIN 0
SERVO_SBUS_RATE  50
SERVO_VOLZ_MASK  0
SIMPLE           0
SR0_ADSB         4
SR0_EXTRA1       4
SR0_EXTRA2       4
SR0_EXTRA3       4
SR0_EXT_STAT     4
SR0_PARAMS       0
SR0_POSITION     4
SR0_RAW_CTRL     4
SR0_RAW_SENS     4
SR0_RC_CHAN      4
SR1_ADSB         0
SR1_EXTRA1       2
SR1_EXTRA2       2
SR1_EXTRA3       0
SR1_EXT_STAT     0
SR1_PARAMS       10
SR1_POSITION     1
SR1_RAW_CTRL     1
SR1_RAW_SENS     1
SR1_RC_CHAN      1
SR2_ADSB         0
SR2_EXTRA1       3
SR2_EXTRA2       2
SR2_EXTRA3       1
SR2_EXT_STAT     2
SR2_PARAMS       0
SR2_POSITION     4
SR2_RAW_CTRL     1
SR2_RAW_SENS     1
SR2_RC_CHAN      1
SR3_ADSB         0
SR3_EXTRA1       0
SR3_EXTRA2       0
SR3_EXTRA3       0
SR3_EXT_STAT     0
SR3_PARAMS       0
SR3_POSITION     0
SR3_RAW_CTRL     0
SR3_RAW_SENS     0
SR3_RC_CHAN      0
SR4_ADSB         0
SR4_EXTRA1       0
SR4_EXTRA2       0
SR4_EXTRA3       0
SR4_EXT_STAT     0
SR4_PARAMS       0
SR4_POSITION     0
SR4_RAW_CTRL     0
SR4_RAW_SENS     0
SR4_RC_CHAN      0
SRTL_ACCURACY    2.000000
SRTL_OPTIONS     0
SRTL_POINTS      300
STAT_BOOTCNT     157
STAT_FLTTIME     24553
STAT_RESET       161089548
STAT_RUNTIME     79598
SUPER_SIMPLE     0
SYSID_ENFORCE    0
SYSID_MYGCS      255
SYSID_THISMAV    1
TCAL_ENABLED     0
TELEM_DELAY      0
TERRAIN_ENABLE   0
THROW_MOT_START  0
THROW_NEXTMODE   18
THROW_TYPE       0
THR_DZ           100
TUNE             0
TUNE_MAX         0.000000
TUNE_MIN         0.000000
VTX_ENABLE       0
WPNAV_ACCEL      300.000000
WPNAV_ACCEL_Z    150.000000
WPNAV_JERK       1.000000
WPNAV_RADIUS     150.000000
WPNAV_RFND_USE   0
WPNAV_SPEED      10000.000000
WPNAV_SPEED_DN   150.000000
WPNAV_SPEED_UP   250.000000
WP_NAVALT_MIN    0.000000
WP_YAW_BEHAVIOR  1
```


```
QGC WPL 110
0	0	0	16	0.000000	0.000000	0.000000	0.000000	0.000000	0.000000	0.000000	1
1	0	3	22	15.000000	0.000000	0.000000	0.000000	0.000000	0.000000	10.000000	1
2	0	3	18	2.000000	0.000000	15.000000	1.000000	44.SCRAMBLED	SCRAMBLED.297341	10.000000	1
3	0	3	16	0.000000	0.000000	0.000000	0.000000	44.SCRAMBLED	SCRAMBLED.297862	13.000000	1
4	0	3	16	0.000000	0.000000	0.000000	0.000000	44.SCRAMBLED	SCRAMBLED.297060	12.000000	1
5	0	0	178	1.000000	14.000000	-1.000000	0.000000	0.000000	0.000000	0.000000	1

```

**Comment by Redshifft** [comment]
![image](https://user-images.githubusercontent.com/11429036/116805599-775f4400-ab1f-11eb-934f-bdc80165b8e2.png)

WD issues from Initialization on 4.1beta, Copter fell out of the sky on board reset after 5 min flight.

WD errors are obviously detected ok so is there any way to start making the Quad "YAW KICK" every 2 seconds or something to get the Pilots attention?

**Comment by andyp1per** [comment]
@Redshifft what flight controller is this?

**Comment by Redshifft** [comment]
Little custom Nano F405 own design. First time it's done this in 60 test flight on 4.1 beta - Fault could be anything from a connection splash to HW/SW 
Like the best Air crashes, Poxy goggles were beeping and I was distracted checking them but nothing obviously wrong while flying. It just fell from  a Hover when the board reset. Logs are incomplete but MP shows the flightpath.
No DMA or over load issues

**Comment by andyp1per** [comment]
I'm just curious because its the infamous "all zeros" watchdog. Still not got to the bottom of what causes this. H7 flash corruption can, but not on an F405.

**Comment by Redshifft** [comment]
I reported it for the reason that it may be possible to add something like I suggested above if detected mid flight.
This is only the second time I've had issue with a F405 board in hundreds of flights. The other one was again at initialization  on a Plane where it set level at 90deg out.
I'm looking into why my arming checks did not pick this up at arm seems a few of the checks were disabled, which is a mystery.

**Comment by Redshifft** [comment]
@andyp1per  I got this wrong, somehow after 5min into flight all this watchdog stuff kicked off, not at bootup. I was quite happy until revisited the evidence. 
After the event I found the SDCard had all this rubbish on it (windows could not see it)
I'll leave it there and we will see what other reports come in.
![SDCard](https://user-images.githubusercontent.com/11429036/116875603-1ca71480-ac13-11eb-804b-bce44ef1d335.png)

**Comment by Notorious811** [comment]
@rmackay9 don't think it is related to the latest firmware release, I'm getting this issue on stable arducopter-4.0.4 and 4.0.7 releases as well, just wanted a little help to decode what's the source

**Comment by bugobliterator** [comment]
@Notorious811 is the issue consistently repeatable?

**Comment by Notorious811** [comment]
> @Notorious811 is the issue consistently repeatable?

yes, it is consistently repeatable with both 4.0 and latest 4.1 arducopter versions of firmware

**Comment by tridge** [comment]
@Notorious811 do you have a real omnibusf4pro you can reproduce it on?

**Comment by peterbarker** [comment]
@Notorious811 can you provide a log - or just run that `WDG` message through `Tools/scripts/decode_watchdog.py` please?

**Comment by Notorious811** [comment]
@tridge this issue is not happening on real omnibusf4pro, it is a custom build FC, close to omnibusf4pro clone just with different sensors, this issue is happening every time after few seconds since powering on the board, it is most probably some design fault from our end but we finding it very difficult to debug, this issue keeps arising even with a minimum hwdef.dat configurations attached below which suggest something we must be doing wrong with setting up components around the MCU.

[hwdef.zip](https://github.com/ArduPilot/ardupilot/files/6578449/hwdef.zip)

**Comment by Notorious811** [comment]
@peterbarker i have multiple logs but as mentioned here https://ardupilot.org/copter/docs/common-watchdog.html#determining-that-a-watchdog-reset-occurred, in my case none of the logs have this info within MSG messages about watchdog reset event, still i'm attaching some of the recent logs-
https://drive.google.com/drive/folders/1qOnV9pUQ03r2o0J1QETysFUKRjHdAnVf?usp=sharing


> just run that `WDG` message through `Tools/scripts/decode_watchdog.py` please

i also tried that but i don't have all the info in proper format about  FA, ICSR etc, that's why this is all the data i able to decode using the image
  T              Scheduler Task:                -2: Fast loop
  IE             Internal Error Mask:           0: ?????
  IEC           Internal Error Count:         0: 0
  IEL            Internal Error Line:            0: 0
  MM          MAVLink Message:           0: [None]
  MC           MAVLink Command:         0: [None]
  SL             Semaphore Line:               0: Not waiting on semaphore
  FL             Fault Line:                      128: ?????
  FT             Fault Type:            3: HardFault

**Comment by estogantsev** [comment]
I've got a bunch of watchdog messages right after landing, all of them were the same:
WDG: T-1 SL0 FL128 FT3 FAE000ED38 FTP180 FLR805F1F

board: omnibusf4pro
fw: plane 4.1.0 beta5

arm switch was still on and dataflash log was being recorded

**Comment by tridge** [comment]
> I've got a bunch of watchdog messages right after landing, all of them were the same:
> WDG: T-1 SL0 FL128 FT3 FAE000ED38 FTP180 FLR805F1F

can you please post the bin logs? There should be 2 logs, one from while it was flying, and a 2nd after the watchdog

**Comment by estogantsev** [comment]
> can you please post the bin logs? There should be 2 logs, one from while it was flying, and a 2nd after the watchdog

attached are 2 log files from that event
[logs.zip](https://github.com/ArduPilot/ardupilot/files/7027656/logs.zip)

**Comment by raj241992** [comment]
I use ArduPlane v 4.1.5 with pixhawk (fmuv3) and I add a serial external AHRS.
when the serial is set in low baud rate speed (<=115200), the code runs fine.
however when I set the baud rate to 460800 the plane code resets and hangs .
if I disconnect the RX pin only ,every thing runs fine and “internal error 0x800” rises
with this msg:

    WDG:T49 SL0 FL129 FT3 FAD7 50C22FTP181 FLR80C7667

if I connect the RX pin after awhile, the sensor works correctly
is there any description to this behavior ? and how can I solve this problem?
best regard.

**Comment by pratikiitgn** [comment]
I have resolved it. Kindly share your query to pkprajapati5959@gmail.com

**Comment by stm32h757** [comment]
ArduRover 4.6.3 (3fc7011a), ground rover on a custom CUAV-X7+-based board (stock CUAV-X7 firmware).
WDG: T18 SL0 FL217 FT4 FA3001CB68 FTP183 FLR12000007 FICSR4196356 MM0 MC0 IE8000 IEC70 TN:moni
→ MemManage, thread moni, main_loop_stuck (IE 0x8000). 

** details, logs and crash_dump.bin here: 
https://discuss.ardupilot.org/t/watchdog-error-crashdump-data-detected-on-ardurover-on-a-custom-board/144108
